velox/functions/prestosql/types/IPAddressType.h

```diff
@@ -15,13 +15,35 @@
  */
 #pragma once
 
+#include <folly/IPAddress.h>
+
 #include "velox/type/SimpleFunctionApi.h"
 #include "velox/type/Type.h"
 
 namespace facebook::velox {
 
+constexpr int kIPV4AddressBytes = 4;
+constexpr int kIPV4ToV6FFIndex = 10;
+constexpr int kIPV4ToV6Index = 12;
+constexpr int kIPAddressBytes = 16;
+
+inline folly::Expected<int128_t, folly::IPAddressFormatError>
+tryGetIPv6asInt128FromString(const std::string& ipAddressStr) {
+  auto maybeIp = folly::IPAddress::tryFromString(ipAddressStr);
+  if (maybeIp.hasError()) {
+    return folly::makeUnexpected(maybeIp.error());
+  }
+
+  int128_t intAddr;
+  folly::IPAddress addr = maybeIp.value();
+  auto addrBytes = folly::IPAddress::createIPv6(addr).toByteArray();
+  std::reverse(addrBytes.begin(), addrBytes.end());
+  memcpy(&intAddr, &addrBytes, kIPAddressBytes);
+  return intAddr;
+}
+
```

Comment:
It's inside the `facebook::velox` ns, I think if we added as part of class IPAddressType as static methods, it may look a bit funky using some of these functions/consts in IPAddressCastOperator class.

Reply:
I put this in its own namespace, I think it'll be a bit cleaner this way 🤷‍♂️